### ISSUE TITLE
Expiração cobrança com vencimento (COBV)

### ISSUE THREAD

**Issue by rudineirk** [bacen/pix-api#523]
Tenho uma dúvida sobre a expiração dos qrcodes de cobrança com vencimento, parece ter alguma divergência das docs com a implementação dos bancos, pois um qrcode que deveria poder ser liquidado estava dando falha na leitura/liquidação por causa do prazo.

Segue o caso:
```json
{
  "calendario": {
    "dataDeVencimento": "2022-08-18",
    "validadeAposVencimento": 2
  },
  ...
}
```

tentando fazer o pagamento desse qrcode na segunda (22/08), ocorreu a seguinte situação nos bancos:
* Nubank: Conseguia ler o qrcode mas dava erro na liquidação
* Santander:  Conseguia ler o qrcode mas dava erro na liquidação
* Inter: Erro de qrcode inválido na leitura
* Bradesco: Erro de qrcode inválido na leitura
* Caixa: Erro "Documento fora do prazo de pagamento" na liquidação do pagamento

pelo que está descrito nessa seção das docs de criação do COBV, deveria ser possível realizar o pagamento desse qrcode:
![Screenshot from 2022-08-22 15-30-29](https://user-images.githubusercontent.com/5260987/186187365-adde4789-01e8-4d48-870a-d68b3c590a5b.png)
![Screenshot from 2022-08-23 11-43-23](https://user-images.githubusercontent.com/5260987/186188188-bd823923-af40-44e8-9330-8a9fbf18b774.png)

A expiração dele seria no sábado (20/08), mas como não é dia útil deveria passar para segunda (22/08).
Qual é o comportamento correto nesses casos? O que os bancos estão implementando ou o que está na documentação?

**Comment by GustavoSTZ** [comment]
Bom dia @rudineirk!

No meu entendimento, o QR Code deve se auto ajustar em tempo de exposição e não em tempo de liquidação.

![image](https://user-images.githubusercontent.com/48111506/186194091-8981fa92-3c1e-41d3-87b1-6fed7eb61bd1.png)

Até porque é comentado na documentação:

> Todos os campos que façam referência a esta data (validadeAposVencimento; desconto; juros e multa) devem assumir essa prorrogação, quando for o caso.

Ou seja, para esses outros campos assumirem tais características, o próprio QR Code deveria se auto ajustar.

**Comment by rubenskuhl** [comment]
> Bom dia @rudineirk!
> 
> No meu entendimento, o QR Code deve se auto ajustar em tempo de exposição e não em tempo de liquidação.
> 
> ![image](https://user-images.githubusercontent.com/48111506/186194091-8981fa92-3c1e-41d3-87b1-6fed7eb61bd1.png)
> 
> Até porque é comentado na documentação:
> 
> > Todos os campos que façam referência a esta data (validadeAposVencimento; desconto; juros e multa) devem assumir essa prorrogação, quando for o caso.
> 
> Ou seja, para esses outros campos assumirem tais características, o próprio QR Code deveria se auto ajustar.

Nope, não é o que diz a especificação. Inclusive pq a informação autoritativa de município, que é integral a fazer essa análise, fica do lado do PSP pagador (sendo passada apenas uma info indicativo no download do payload). É por causa de interpretações como essa que um QR-Code correto está sendo rejeitado, o que vai obrigar a ações mais firmes do regulador. 

@rudineirk eu abriria RDRs contra conta uma dessas instituições que está agindo incorretamente.

**Comment by rubenskuhl** [comment]
@thiagolvlsantos @ninrod

**Comment by thiagolvlsantos** [comment]
Prezados,

A função básica da ApiPix é disciplinar o comportamento esperado dos provedores, inclusive com os exemplos para minimizar as eventuais falhas de interpretação. Desta forma, @rudineirk, sim, sua interpretação está correta inclusive de acordo com o exemplo citado.

Qualquer não-conformidade, por parte dos provedores com as especificações e exemplos na ApiPix, deve ser reportada ao pix-operacional@bcb.gov.br (e)ou abrir RDRs para devida apuração.

Obs.: Caso a resposta seja satisfatória, pede-se a gentileza de finalizar a issue.

Atenciosamente,
Thiago Santos

**Comment by rudineirk** [comment]
Obrigado pelos retornos, vamos tentar encaminhar esse problema para os contatos comentados.
